nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -107,42 +124,102 @@ def output_types(self):
             {
                 "outputs": NeuralType(('B', 'D', 'T'), AcousticEncodedRepresentation()),
                 "encoded_lengths": NeuralType(tuple('B'), LengthsType()),
+                "cache_last_channel_next": NeuralType(('D', 'B', 'T', 'D'), ChannelType(), optional=True),
+                "cache_last_time_next": NeuralType(('D', 'B', 'D', 'T'), ChannelType(), optional=True),
             }
         )
 
+    @property
+    def disabled_deployment_input_names(self):
+        if not self.export_cache_support:
+            return set(["cache_last_channel", "cache_last_time"])
+        else:
+            return set()
+
+    @property
+    def disabled_deployment_output_names(self):
+        if not self.export_cache_support:
+            return set(["cache_last_channel_next", "cache_last_time_next"])
+        else:
+            return set()
+
     def __init__(
         self,
         feat_in,
         n_layers,
         d_model,
         feat_out=-1,
+        causal_downsampling=False,
         subsampling='striding',
         subsampling_factor=4,
         subsampling_conv_channels=-1,
         ff_expansion_factor=4,
         self_attention_model='rel_pos',
         n_heads=4,
         att_context_size=None,
+        att_context_style='regular',
         xscaling=True,
         untie_biases=True,
         pos_emb_max_len=5000,
         conv_kernel_size=31,
         conv_norm_type='batch_norm',
+        conv_context_size=None,
         dropout=0.1,
         dropout_emb=0.1,
         dropout_att=0.0,
     ):
         super().__init__()
-
         d_ff = d_model * ff_expansion_factor
         self.d_model = d_model
+        self.n_layers = n_layers
         self._feat_in = feat_in
         self.scale = math.sqrt(self.d_model)
+        self.att_context_style = att_context_style
```

Comment:
Assert possible values of this  argument

Reply:
there is a block some lines after where the valid values are checked and raise an error if needed.